tests/integration/ApiClient/IdempotencyKeyTest.php

```diff
@@ -52,3 +52,53 @@
     $error = ErrorResponse::idempotencyKeyReuse();
     \request([$error, $error, $error])->post();
 })->throws(ApiResponseUnsuccessfulException::class);
+
+\it('uses custom idempotency key when provided', function () {
+    $requestOptions = \client()
+        ->requestOptions()
+        ->idempotencyKey('test-idempotency-key');
+
+    \request()
+        ->requestOptions($requestOptions)
+        ->post();
+
+    expect(getRequestIdempotencyKey(1))->toBe('test-idempotency-key');
```

Comment:
Is it worth also asserting that there is only one idempotency header on the request?